### PR TITLE
Contests Index

### PR DESCRIPTION
**Pull request by nekodex**
ayup.

### CODE REVIEW

File: resources/assets/less/bem/contest-list-item.less [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  &__right-content {
+    float: right;
+    padding: @spacing;
+    flex-shrink: 0;
+  }
+
+  &--preparing {
+    border-right: 5px solid @yellow;
+    &:hover {
+      border-right: 5px solid lighten(@yellow, 5%);
```
Comment (nanaya) [inline resources/assets/less/bem/contest-list-item.less]: I think `border-color` is the one you're looking for.

File: resources/lang/en/contest.php [hunk 2]
```diff
@@ -19,6 +19,10 @@
  */
 
 return [
+    'header' => [
+        'small' => 'Community Contests and Stuff',
```
Comment (Ephemeralis) [inline resources/lang/en/contest.php]: "Contests of all kinds, open to the community."

File: app/helpers.php [hunk 3]
```diff
@@ -379,6 +380,11 @@ function display_regdate($user)
     return trans('users.show.joined_at', ['date' => $user->user_regdate->formatLocalized('%B %Y')]);
 }
 
+function i18n_date($datetime, $format = 'dd MMMM yyyy')
```
Comment (nanaya) [inline app/helpers.php]: if it's just date, `[IntlDateFormatter::LONG, IntlDateFormatter::NONE]` is better...

...is what I want to say but for some reason it doesn't work with `::NONE` format for time. This does work, though:

```php
$formatter = IntlDateFormatter::create(
    App::getLocale(),
    IntlDateFormatter::LONG,
    IntlDateFormatter::NONE
);
$formatter->format($datetime);
```

File: resources/views/artists/index.blade.php [hunk 4]
```diff
@@ -27,8 +27,8 @@
     <div class="osu-layout__row">
         <div class="osu-page-header-v2 osu-page-header-v2--featured-artists">
             <div class="osu-page-header-v2__overlay"></div>
-            <div class="osu-page-header-v2__subtitle osu-page-header-v2__subtitle--artist">{{trans('artist.beatmaps._')}} &raquo;</div>
             <div class="osu-page-header-v2__title">{{trans('artist.title')}}</div>
+            <div class="osu-page-header-v2__subtitle"></div>
```
Comment (nanaya) [inline resources/views/artists/index.blade.php]: doesn't seem to be doing anything... (also for other pages)

File: resources/assets/less/bem/contest-list-legend.less [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  display: flex;
+  justify-content: flex-end;
+
+  &__item {
+    display: inline-block;
+    height: 30px;
+    background: #323E41;
+    padding: @spacing / 2;
+    margin: @spacing / 2;
+    padding-left: @spacing;
```
Comment (nanaya) [inline resources/assets/less/bem/contest-list-legend.less]: more like `padding: (@spacing / 2) @padding;`

**Pull request merged**